File: python/acronym/acronym.py
import re


def find_uppercases(word):
    yield word[0].upper()
    for character in word[1:]:
        if character.isupper():
            yield character

def abbreviate(name):
    split_name = re.split(r'[ -]', name)

    output = []
    for word in split_name:
        if word.isupper():
            output.append(word[0])
            continue

        output.extend(find_uppercases(word))

    return ''.join(output)

File: javascript/acronym/acronym.js
function Acronym() {}

Acronym.prototype.parse = function(name) {
    var split_name = name.replace(/[,]/).split(/[- ]/);
    
    var acronym_output = "";
    var split_name_len = split_name.length;
    for (var i = 0; i < split_name_len; ++i) {
        if (split_name[i] == split_name[i].toUpperCase()) {
            acronym_output += split_name[i][0];
            continue;
        }

        var word = split_name[i];
        acronym_output += word[0];
        var word_len = word.length;
        for (var j = 1; j < word_len; ++j) {
            if (word[j] == word[j].toUpperCase()) {
                acronym_output += word[j];
            }
        }
    }

    return acronym_output.toUpperCase();
};

module.exports = new Acronym();